packages/patternfly-4/react-core/src/components/Dropdown/DropdownItem.js

```diff
@@ -41,6 +41,7 @@ const DropdownItem = ({ className, children, isHovered, component: Component, is
       <Component
         {...additionalProps}
         className={css(isDisabled && styles.modifiers.disabled, isHovered && styles.modifiers.hover, className)}
+        onClick={onSelect}
```

Comment:
There is a bit of an issue with always setting `onClick` to `onSelect` here.  If the dropdown has multiple button items, this will set all of their `onClick` handlers to `onSelect` (so all buttons will fire the same action).

One possible solution would be to add a `handleOnClick` handler to `DropdownItem` and do something like the following

```js
// ...
return (
  <li>
    <Component
      {...additionalProps}
      className={css(isDisabled && styles.modifiers.disabled, isHovered && styles.modifiers.hover, className)}
      onClick={event => handleOnClick && handleOnClick(event, onSelect)}
    >
      {children}
    </Component>
  </li>
);
```

Then the consuming app could do something like

```js
<DropdownItem
  // ...
  component="button"
  handleOnClick={(event, onSelect) => {
    // fire off the particular action for this button
    onSelect();
  }}
>
// ...
</DropdownItem>
```

But this would mean that all `DropdownItem`s (even the "link" ones) will require an `onClick` handler to ensure that `onSelect` is fired.